src/util/client-info.ts

```diff
@@ -0,0 +1,42 @@
+import assign from "../util/assign";
+import "unfetch/polyfill";
+
+interface Cache {
+  [key: string]: Promise<ClientInfo>;
+}
+
+const cache: Cache = {};
+
+/** Cleans the cache: necessary for testing */
+function reset(url: string): void {
+  delete cache[url];
+}
+
+/* eslint-disable @typescript-eslint/camelcase */
+function normalizeResponse(raw: RawClientInfo): ClientInfo {
+  return assign({}, raw, {
+    client_asn: parseInt(raw.client_asn, 10),
+    resolver_asn: parseInt(raw.resolver_asn, 10)
+  });
+}
```

Comment:
I'd like to get rid of this and ensure that the server sends the correct type rather than patching on the client.